### PR TITLE
Migrate from winapi to windows-sys

### PR DESCRIPTION
**Pull request by killianc3**
After seeing this issue #5993 and because winapi is not maintained, i think migrating to windows-sys is a good thing, what do you think ?

### CODE REVIEW

File: alacritty_terminal/src/tty/windows/conpty.rs [hunk 1]
```diff
@@ -134,11 +134,11 @@ pub fn new(config: &PtyConfig, window_size: WindowSize) -> Option<Pty> {
     let cmdline = win32_string(&cmdline(config));
     let cwd = config.working_directory.as_ref().map(win32_string);
 
-    let mut proc_info: PROCESS_INFORMATION = Default::default();
+    let mut proc_info: PROCESS_INFORMATION = unsafe { mem::zeroed() };
```
Comment (chrisduerr) [inline alacritty_terminal/src/tty/windows/conpty.rs:137]: Same safety concern here, how did you validate this call's safety?

**Review by chrisduerr — CHANGES_REQUESTED** [review]

File: alacritty_terminal/src/tty/windows/conpty.rs [hunk 2]
```diff
@@ -65,11 +65,11 @@ pub fn new(config: &PtyConfig, window_size: WindowSize) -> Option<Pty> {
 
     // Prepare child process startup info.
 
-    let mut size: SIZE_T = 0;
+    let mut size: usize = 0;
 
-    let mut startup_info_ex: STARTUPINFOEXW = Default::default();
+    let mut startup_info_ex: STARTUPINFOEXW = unsafe { mem::zeroed() };
```
[the remark this answers is not in the document]
Comment (killianc3) [reply alacritty_terminal/src/tty/windows/conpty.rs:70]: I'm not sure but if you check in wanapi, they implement Default::default() exactly like that.
https://github.com/retep998/winapi-rs/blob/2f76bdea3a79817ccfab496fbd1786d5a697387b/src/macros.rs#L292

File: alacritty_terminal/src/tty/windows/child.rs [hunk 3]
```diff
@@ -29,15 +31,15 @@ impl ChildExitWatcher {
     pub fn new(child_handle: HANDLE) -> Result<ChildExitWatcher, Error> {
         let (event_tx, event_rx) = channel::<ChildEvent>();
 
-        let mut wait_handle: HANDLE = 0 as HANDLE;
+        let mut wait_handle = 0isize;
```
[the remark this answers is not in the document]
Comment (chrisduerr) [reply alacritty_terminal/src/tty/windows/child.rs:34]: Please don't just resolve the conversation without at the very least replying to it.

File: alacritty_terminal/src/tty/windows/conpty.rs [hunk 4]
```diff
@@ -39,7 +39,7 @@ impl Drop for Conpty {
 unsafe impl Send for Conpty {}
 
 pub fn new(config: &PtyConfig, window_size: WindowSize) -> Option<Pty> {
-    let mut pty_handle = 0 as HPCON;
+    let mut pty_handle = 0isize;
```
[the remark this answers is not in the document]
Comment (chrisduerr) [reply alacritty_terminal/src/tty/windows/conpty.rs:42]: Please don't just resolve the conversation without at the very least replying to it.

File: alacritty_terminal/src/tty/windows/conpty.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (chrisduerr) [reply alacritty_terminal/src/tty/windows/conpty.rs:70]: Yeah I saw that, just to be clear I checked the struct myself and didn't find any references or function pointers or anything like that stored in it. So I *think* it should be safe. Just wanted to know if you checked it.

File: alacritty_terminal/src/tty/windows/child.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (killianc3) [reply alacritty_terminal/src/tty/windows/child.rs:34]: yep, sorry this is my first pr here. (and I agree with these changes)

File: alacritty_terminal/src/tty/windows/conpty.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (killianc3) [reply alacritty_terminal/src/tty/windows/conpty.rs:42]: yep, sorry this is my first pr here. (and I agree with these changes)

File: alacritty_terminal/src/tty/windows/child.rs (hunk 3 above)
Comment (kchibisov) [inline alacritty_terminal/src/tty/windows/child.rs:34]: Could you use proper type alias? There's `HANDLE` in foundation.

File: alacritty_terminal/src/tty/windows/child.rs [hunk 5]
```diff
@@ -4,14 +4,16 @@ use std::sync::atomic::{AtomicPtr, Ordering};
...
+use windows_sys::Win32::System::Threading::{
+    RegisterWaitForSingleObject, UnregisterWait, WT_EXECUTEINWAITTHREAD, WT_EXECUTEONLYONCE,
+};
+use windows_sys::Win32::System::WindowsProgramming::INFINITE;
 
 use crate::tty::ChildEvent;
 
 /// WinAPI callback to run when child process exits.
-extern "system" fn child_exit_callback(ctx: PVOID, timed_out: BOOLEAN) {
+extern "system" fn child_exit_callback(ctx: *mut c_void, timed_out: BOOLEAN) {
```
Comment (kchibisov) [inline alacritty_terminal/src/tty/windows/child.rs:16]: Are you sure that there's no special type for that on windows? What native apps use here instead?

File: alacritty_terminal/src/tty/windows/conpty.rs (hunk 4 above)
Comment (kchibisov) [inline alacritty_terminal/src/tty/windows/conpty.rs:42]: Ditto wrt `HPCON` it's present in windows-sys as well.

File: alacritty/src/display/window.rs [hunk 6]
```diff
@@ -63,7 +61,7 @@ static WINDOW_ICON: &[u8] = include_bytes!("../../extra/logo/compat/alacritty-te
 
 /// This should match the definition of IDI_ICON from `alacritty.rc`.
 #[cfg(windows)]
-const IDI_ICON: WORD = 0x101;
+const IDI_ICON: u16 = 0x101;
```
Comment (kchibisov) [inline alacritty/src/display/window.rs:64]: I guess windows-sys don't have aliases for those?

**Review by kchibisov — DISMISSED** [review]
Could you check that we use the same windows-sys across all the deps? I think winit is using some other value for that.

File: alacritty_terminal/src/tty/windows/child.rs (hunk 5 above)
Comment (killianc3) [reply alacritty_terminal/src/tty/windows/child.rs:16]: Yes, looking at the Windows-Sys documentation, this is how they defined this function. (https://docs.rs/windows-sys/latest/windows_sys/Win32/System/Threading/type.WAITORTIMERCALLBACK.html)

File: alacritty_terminal/src/tty/windows/child.rs (hunk 3 above)
Comment (killianc3) [reply alacritty_terminal/src/tty/windows/child.rs:34]: Yes, I'll do it this way, it's better.

File: alacritty/src/display/window.rs (hunk 6 above)
Comment (killianc3) [reply alacritty/src/display/window.rs:64]: That's right! I didn't find this in the documentation, the closest is PWSTR which is *const u16.

File: alacritty_terminal/src/tty/windows/child.rs [hunk 7]
```diff
@@ -29,15 +31,15 @@ impl ChildExitWatcher {
...
+        let mut wait_handle = 0isize;
         let sender_ref = Box::new(event_tx);
 
         let success = unsafe {
             RegisterWaitForSingleObject(
                 &mut wait_handle,
                 child_handle,
                 Some(child_exit_callback),
-                Box::into_raw(sender_ref) as PVOID,
+                Box::into_raw(sender_ref) as *mut c_void,
```
[the remark this answers is not in the document]
Comment (killianc3) [reply alacritty_terminal/src/tty/windows/child.rs:42]: Same, I'll do it this way, it's better.

File: alacritty_terminal/src/tty/windows/conpty.rs (hunk 4 above)
Comment (killianc3) [reply alacritty_terminal/src/tty/windows/conpty.rs:42]: Same, I'll do it this way, it's better.

**Comment by killianc3** [comment]
Yes, after checking, winit uses version 0.36 for windows-sys so I configured it the same way.

**Review by chrisduerr — APPROVED** [review]
LGTM

**Pull request merged**